tests/test_validators.py

```diff
@@ -173,6 +173,21 @@ def test_GIVEN_valid_string_value_WHEN_validating_field_value_THEN_returns_accep
     validator.is_valid.emit.assert_called_once_with(True)
 
 
+def test_GIVEN_valid_name_value_WHEN_validating_filewriter_THEN_returns_acceptable_and_emits_signal_with_true():
+    strvalue = "123a"
+
+    field_type_combo = DummyCombo(FieldType.filewriter.value)
+    dataset_type_combo = DummyCombo(ValueTypes.STRING)
+
+    validator = FieldValueValidator(
+        field_type_combo, dataset_type_combo, FieldType.filewriter.value
+    )
+    validator.is_valid = Mock()
+
+    assert validator.validate(strvalue, 0) == QValidator.Acceptable
+    validator.is_valid.emit.assert_called_once_with(True)
+
+
 def test_GIVEN_invalid_float_value_WHEN_validating_field_value_THEN_returns_intermediate_and_emits_signal_with_false():
     invalid_value = "sdfn"
```

Comment:
Agreed, I think one of the things I will do during the restructure is to ensure better test coverage.  UI checks failed on macOS due to the adjustment for dark mode, will chat with Afonso when he's here about how exactly the environments are set but in principle the OS should have some flag that Qt accesses to check for dark mode

Reply:
I think it might be worthwhile making another ticket for test coverage, as currently none of the field types are checked (and other issues exist too).  I've added it to a JSON string though to make sure it doesn't at least cause issues on import.